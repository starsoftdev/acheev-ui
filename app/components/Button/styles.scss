@import '~styles/settings/settings';

// Foundation for Sites by ZURB
// foundation.zurb.com
// Licensed under MIT Open Source

////
/// @group button
////

/// Padding inside buttons.
/// @type List
$button-padding: 0.85em 1em !default;

/// Margin around buttons.
/// @type List
$button-margin: 0 0 $global-margin 0 !default;

/// Default fill for buttons. Can either be `solid` or `hollow`.
/// @type Keyword
$button-fill: solid !default;

/// Default background color for buttons.
/// @type Color
$button-background: $primary-color !default;

/// Background color on hover for buttons.
/// @type Color
$button-background-hover: scale-color(
  $button-background,
  $lightness: -15%
) !default;

/// Font color for buttons.
/// @type List
$button-color: $white !default;

/// Alternative font color for buttons.
/// @type List
$button-color-alt: $black !default;

/// Border radius for buttons, defaulted to global-radius.
/// @type Number
$button-radius: $global-radius !default;

/// Border width for hollow outline buttons
/// @type Number
$button-hollow-border-width: 1px !default;

/// Coloring classes. A map of classes to output in your CSS, like `.secondary`, `.success`, and so on.
/// @type Map
$button-palette: $foundation-palette !default;

/// opacity for a disabled button.
/// @type List
$button-opacity-disabled: 0.25 !default;

/// Background color lightness on hover for buttons.
/// @type Number
$button-background-hover-lightness: -20% !default;

/// Color lightness on hover for hollow buttons.
/// @type Number
$button-hollow-hover-lightness: -50% !default;

// Internal: flip from margin-right to margin-left for defaults
@if $global-text-direction == 'rtl' {
  $button-margin: 0 0 $global-margin $global-margin !default;
}

/// transitions for buttons.
/// @type List
$button-transition: background-color 0.25s ease-out, color 0.25s ease-out !default;

// TODO: Document button-base() mixin
@mixin button-base {
  @include disable-mouse-outline;
  margin: $button-margin;

  -webkit-appearance: none;
  border: 1px solid transparent;
  border-radius: $button-radius;
  transition: $button-transition;
  display: inline-block;

  font-size: map-deep-get($button-sizes, default, font-size);
  padding: map-deep-get($button-sizes, default, padding);
  font-weight: bold;
  text-align: center;
  text-transform: none;
  cursor: pointer;
  text-decoration: none;

  white-space: nowrap;
}

/// Expands a button to make it full-width.
/// @param {Boolean} $expand [true] - Set to `true` to enable the expand behavior. Set to `false` to reverse this behavior.
@mixin button-expand($expand: true) {
  @if $expand {
    display: block;
    width: 100%;
    margin-right: 0;
    margin-left: 0;
  } @else {
    display: inline-block;
    width: auto;
    margin: $button-margin;
  }
}

/// Removes background fill on hover and focus for hollow buttons.
@mixin button-hollow {
  &,
  &:hover,
  &:focus {
    background-color: transparent;
  }
}

@mixin button-hollow-style(
  $color: $primary-color,
  $hover-lightness: $button-hollow-hover-lightness,
  $border-width: $button-hollow-border-width
) {
  $color-hover: scale-color($color, $lightness: $hover-lightness);

  border: $border-width solid $color;
  color: $color;

  &:hover,
  &:focus {
    color: $color-hover;
  }
}

/// Adds disabled styles to a button by fading the element, reseting the cursor, and disabling pointer events.
/// @param [Color] $background [$primary-color] - Background color of the disabled button.
/// @param [Color] $color [$button-color] - Text color of the disabled button. Set to `auto` to have the mixin automatically generate a color based on the background color.
@mixin button-disabled($background: $primary-color, $color: $button-color) {
  @if $color == auto {
    $color: color-pick-contrast(
      $background,
      ($button-color, $button-color-alt)
    );
  }

  opacity: $button-opacity-disabled;
  cursor: not-allowed;

  &,
  &:hover,
  &:focus {
    background-color: $background;
    color: $color;
  }
}

/// Adds a dropdown arrow to a button.
/// @param {Number} $size [0.4em] - Size of the arrow. We recommend using an `em` value so the triangle scales when used inside different sizes of buttons.
/// @param {Color} $color [white] - Color of the arrow.
/// @param {Number} $offset [$button-padding] - Distance between the arrow and the text of the button. Defaults to whatever the right padding of a button is.
@mixin button-dropdown(
  $size: 0.4em,
  $color: $white,
  $offset: get-side($button-padding, right)
) {
  &::after {
    @include css-triangle($size, $color, down);
    position: relative;
    top: 0.4em; // Aligns the arrow with the text of the button

    display: inline-block;
    float: #{$global-right};
    margin-#{$global-left}: get-side($button-padding, right);
  }
}

/// Adds all styles for a button. For more granular control over styles, use the individual button mixins.
/// @param {Boolean} $expand [false] - Set to `true` to make the button full-width.
/// @param {Color} $background [$button-background] - Background color of the button.
/// @param {Color} $background-hover [$button-background-hover] - Background color of the button on hover. Set to `auto` to have the mixin automatically generate a hover color.
/// @param {Color} $color [$button-color] - Text color of the button. Set to `auto` to automatically generate a color based on the background color.
/// @param {Keyword} $style [solid] - Set to `hollow` to create a hollow button. The color defined in `$background` will be used as the primary color of the button.
@mixin button(
  $expand: false,
  $background: $button-background,
  $background-hover: $button-background-hover,
  $color: $button-color,
  $style: $button-fill
) {
  @include button-base;

  @if $style == solid {
    @include button-style($background, $background-hover, $color);
  } @else if $style == hollow {
    @include button-hollow;
    @include button-hollow-style($background);
  }

  @if $expand {
    @include button-expand;
  }
}

@mixin foundation-button {
  .button {
    @include button;

    &__spinner {
      display: inline-block;
      margin: rem-calc(-6) $rem-sm rem-calc(-2);
    }

    // Sizes
    @each $size, $value in map-remove($button-sizes, default) {
      &.#{$size} {
        font-size: map-get($value, font-size);
        padding: map-get($value, padding);
      }
    }

    &.expanded {
      @include button-expand;
    }

    // Colors
    @each $name, $color in $button-palette {
      @if $button-fill != hollow {
        &.#{$name} {
          @include button-style($color, auto, auto);
        }
      } @else {
        &.#{$name} {
          @include button-hollow-style($color);
        }

        &.#{$name}.dropdown::after {
          border-top-color: $color;
        }
      }
    }

    // Hollow style
    @if $button-fill != hollow {
      &.hollow {
        @include button-hollow;
        @include button-hollow-style;

        @each $name, $color in $button-palette {
          &.#{$name} {
            @include button-hollow-style($color);
          }
        }
      }
    }

    // Disabled style
    &.disabled,
    &[disabled] {
      @include button-disabled;

      @each $name, $color in $button-palette {
        &.#{$name} {
          @include button-disabled($color, auto);
        }
      }
    }

    // Dropdown arrow
    &.dropdown {
      @include button-dropdown;

      @if $button-fill == hollow {
        &::after {
          border-top-color: $button-background;
        }
      }

      &.hollow {
        &::after {
          border-top-color: $button-background;
        }

        @each $name, $color in $button-palette {
          &.#{$name} {
            &::after {
              border-top-color: $color;
            }
          }
        }
      }
    }

    &.shadow {
      box-shadow: 0 20px 40px rgba(0, 0, 0, 0.175);
    }

    &.light {
      color: $darker-gray !important;
    }

    &.active {
      // looks the same as dark. needed for active button in a button group
      &,
      &:hover,
      &:focus {
        background-color: $darker-gray;
        border-color: $darker-gray !important;
        color: $white !important;
      }
    }

    @at-root .button-group.air & {
      text-transform: none;
      font-size: $fs-md !important;
      color: $dark-gray !important;
      border-color: $secondary;
      border-top: 0;
      border-bottom: 0;
      padding-top: rem-calc(8);
      padding-bottom: rem-calc(8);
      &:first-child {
        border-left: 0;
      }
      &:last-child {
        border-right: 0;
      }
      &.active,
      &:hover {
        background-color: $white;
        color: $secondary !important;
      }
    }

    &.someSpacious {
      padding-left: 3em;
      padding-right: 3em;
    }
    &.semiSpacious {
      padding-left: 5em;
      padding-right: 5em;
    }
    &.spacious {
      padding-left: 7em;
      padding-right: 7em;
    }

    @each $size in $breakpoint-classes {
      &.#{$size}-semiSpacious {
        @include breakpoint($size) {
          padding-left: 5em;
          padding-right: 5em;
        }
      }
      &.#{$size}-spacious {
        @include breakpoint($size) {
          padding-left: 7em;
          padding-right: 7em;
        }
      }
    }

    // Button with dropdown arrow only
    &.arrow-only::after {
      top: -0.1em;
      float: none;
      margin-#{$global-left}: 0;
    }
  }
}

@include foundation-button;
