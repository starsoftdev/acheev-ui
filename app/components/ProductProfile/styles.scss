@import '~styles/settings/settings';

.productProfile {
  padding-bottom: $rem-xl;
  margin-bottom: $rem-xl;
  @include border;
  @include breakpoint(small only) {
    padding-bottom: $rem-md;
    margin-bottom: $rem-md;
  }

  &__thumbnail {
    width: 170px;
    height: 170px;
    border-radius: 50%;
    background-position: center;
    background-size: cover;
    border: 1px solid $light-gray;
    @include breakpoint(small only) {
      width: 75px;
      height: 75px;
    }
  }
  &__producerName {
    margin-bottom: $rem-md;
    text-transform: uppercase;
    color: $dark-gray;
    font-weight: bold;
  }
  &__producerNameLink {
    color: $dark-gray;
    text-decoration: none;
    font-weight: bold;
    text-transform: uppercase;
    font-size: $fs-base;
    &:hover {
      color: $dark-gray;
    }
  }
  &__price {
    margin: 0;
    font-weight: 300;
    font-size: $rem-md;
    color: $darker-gray;
  }
  &__points {
    font-weight: 300;
    color: $dark-gray;
  }
}

.addToBagButton {
  background-color: #a0ce67;
}
