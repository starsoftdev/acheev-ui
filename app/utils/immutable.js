// @flow

import { Iterable } from 'immutable';
import type { Map } from 'immutable';
import { get as _get } from 'lodash-es';

export const get = (
  collection: ?Map<*, *> | Object,
  path: string | number,
  defaultValue: any
) => {
  if (!collection) {
    return defaultValue;
  } else if (Iterable.isIterable(collection)) {
    return collection.get(path, defaultValue);
  }
  return _get(collection, path, defaultValue);
};

export const getIn = (
  collection: ?Map<*, *> | Object,
  path: Array<string | number>,
  defaultValue: any
) => {
  if (!collection) {
    return defaultValue;
  } else if (Iterable.isIterable(collection)) {
    return collection.getIn(path, defaultValue);
  }
  return _get(collection, path, defaultValue);
};
