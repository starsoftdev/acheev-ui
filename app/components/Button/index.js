// @flow

import * as React from 'react';
import cx from 'classnames';

import Spinner from 'components/Spinner';
import Link from 'components/Link';

import './styles.scss';

type Props = {
  element?: any,
  className?: string,
  children?: any,
  onClick?: Function,
  to?: string,
  type?: string,
  activeClassName?: string,
  disabled?: boolean,
  isLoading?: boolean,
  exact?: boolean,
};

const Button = (props: Props) => {
  const {
    className,
    element = 'a',
    children,
    onClick,
    to,
    type,
    activeClassName,
    disabled,
    isLoading,
    exact,
  } = props;
  const mergedClassName = cx('button', className);
  const actualProps: Object = {
    className: mergedClassName,
    onClick,
    to,
    disabled: disabled || isLoading,
  };
  if (activeClassName) actualProps.activeClassName = activeClassName;
  if (element === 'button') actualProps.type = type;
  if (element === Link && exact) actualProps.exact = exact;
  return React.createElement(
    type ? 'button' : element,
    actualProps,
    <div className="row align-middle align-center">
      {isLoading && (
        <div className="shrink column">
          <Spinner className="button__spinner" />
        </div>
      )}
      <div className="shrink column">{[children]}</div>
    </div>
  );
};

export default Button;
