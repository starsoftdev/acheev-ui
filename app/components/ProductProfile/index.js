// @flow

import React, { Component } from 'react';
import Link from 'components/Link';
import type { Map } from 'immutable';
import cx from 'classnames';

import Button from 'components/Button';
import ReviewCount from 'components/ReviewCount';
import ProductPrice from 'components/ProductPrice';
import ProductAction from 'components/ProductAction';
import ProductGallery from 'components/ProductGallery';
import CustomSelect from 'components/CustomSelect';
import RequireAuth from 'components/RequireAuth';

import transformOptions from 'utils/transformOptions';
import pluralizeCategory from 'utils/pluralizeCategory';

import ProductCategory from 'components/ProductCategory';
import Cannabinoids from 'components/Cannabinoids';

import IconPlus from 'images/sprite/plus.svg';
import IconHeart from 'images/sprite/heart.svg';
import './styles.scss';

const quantities = ['1', '2', '3', '4', '5'];

type Props = {
  data: Map<string, any>,
  addToCart: Function,
  shopifyProduct: ?Object,
  followLikeProduct: Function,
  currentUser: Object,
  follows: Object,
  likes: Object,
};
type State = {
  quantity: string,
  product: ?Object,
};

class ProductProfile extends Component<Props, State> {
  static getDerivedStateFromProps(nextProps: Props) {
    const { shopifyProduct } = nextProps;
    return {
      product: shopifyProduct,
    };
  }
  constructor(props: Props) {
    super(props);
    this.state = {
      quantity: '1',
      product: props.shopifyProduct,
    };
  }
  followLikeProduct = (actionType: string) => {
    this.props.followLikeProduct(this.props.data.get('_id'), actionType);
  };
  render() {
    const { data, addToCart, follows, likes, currentUser } = this.props;
    if (!data) return null;
    const { quantity, product } = this.state;

    const name = data.get('name');
    const photos = data.get('photos');
    const producer = data.getIn(['business', 'name']);
    const businessCategory = data.getIn(['business', '__t']);
    const businessCategoryPlural =
      businessCategory && pluralizeCategory(businessCategory);
    const businessSlug = data.getIn(['business', 'slug']);
    const productCategory = data.get('__t');
    const categoryPlural = pluralizeCategory(productCategory);
    const slug = data.get('slug');
    const price = data.getIn(['variants', 0, 'price']);
    const priceInfo = (
      <div>
        <span className="productProfile__price">
          <ProductPrice
            price={price}
            unit={data.getIn(['variants', '0', 'doseUnit'])}
            doseAmount={data.getIn(['variants', '0', 'doseAmount'])}
          />
        </span>
      </div>
    );
    const thumbnail =
      data.get('thumbnail') || data.getIn(['business', 'thumbnail']);
    return (
      <div className="productProfile row">
        <div
          className={cx(
            'small-12 columns',
            this.props.shopifyProduct ? 'medium-6' : 'medium-4'
          )}
        >
          {photos.size ? (
            <ProductGallery data={photos} title={name} />
          ) : (
            <div
              className="productProfile__thumbnail mb-md"
              style={{ backgroundImage: `url(${String(thumbnail)})` }}
            />
          )}
        </div>
        <div
          className={cx(
            'small-12 columns',
            this.props.shopifyProduct ? 'medium-6' : 'medium-8'
          )}
        >
          <h2 className="c-darkest-gray">{name}</h2>
          {producer &&
            businessCategoryPlural &&
            businessSlug && (
              <div className="productProfile__producerName">
                <Link
                  className="productProfile__producerNameLink"
                  to={`/${businessCategoryPlural}/${businessSlug}`}
                >
                  by {producer}
                </Link>
              </div>
            )}
          <div className="row">
            {data.get('category') && (
              <div className="shrink column">
                <ProductCategory data={data.get('category')} />
              </div>
            )}
            <div className="shrink column">
              <Cannabinoids
                thc={data.get('thcHigh')}
                cbd={data.get('cbdHigh')}
                category={productCategory}
              />
            </div>
          </div>
          <div className="row align-middle mb-md">
            <div className="column small-12 large-expand mb-md">
              <ReviewCount
                reviewCount={data.get('reviews').size}
                ratingsAverage={data.get('rating')}
                to={`/${categoryPlural}/${slug}/reviews`}
              />
            </div>
            <div className="shrink column mb-md">
              <Link
                to={`/${categoryPlural}/${slug}/create-review`}
                className="button secondary"
              >
                Review
              </Link>
            </div>
          </div>
          <div className="row align-middle mb-md">
            <div className="column">{priceInfo}</div>
            <div className="shrink column right">
              {currentUser ? (
                <ProductAction
                  glyph={IconPlus}
                  title="Follow"
                  count={follows && follows.get('count')}
                  onClick={() => this.followLikeProduct('follow')}
                />
              ) : (
                <RequireAuth toDo="follow products">
                  <ProductAction
                    glyph={IconPlus}
                    title="Follow"
                    count={follows && follows.get('count')}
                  />
                </RequireAuth>
              )}
            </div>
            <div className="shrink column">
              {currentUser ? (
                <ProductAction
                  glyph={IconHeart}
                  title="Like"
                  count={likes && likes.get('count')}
                  onClick={() => this.followLikeProduct('like')}
                />
              ) : (
                <RequireAuth toDo="like product">
                  <ProductAction
                    glyph={IconHeart}
                    title="Like"
                    count={likes && likes.get('count')}
                  />
                </RequireAuth>
              )}
            </div>
          </div>
          {product &&
            product.options.map(option => {
              if (option.values.length <= 1) {
                // products with no variants should not show the dropdown UI
                // see related: `var variant_title` and 'Default Title'
                return '';
              }
              return (
                <div className="row align-middle mb-sm" key={option.name}>
                  <div className="shrink column npr mb-sm">
                    <label htmlFor={option.name}>{option.name}</label>
                  </div>
                  <div className="column mb-sm">
                    <CustomSelect
                      className="large"
                      name={option.name}
                      options={transformOptions(option.values)}
                      clearable={false}
                      value={option.selected}
                      onChange={e => {
                        product.options.filter(
                          filteredOption => filteredOption.name === option.name
                        )[0].selected =
                          e.value;
                        this.setState({ product });
                      }}
                    />
                  </div>
                </div>
              );
            })}
          {product && (
            <div className="row align-middle">
              <div className="shrink column mb-md">
                <label htmlFor="qty">Quantity</label>
              </div>
              <div className="small-expand medium-shrink column mb-md">
                <CustomSelect
                  className="large"
                  name="qty"
                  options={transformOptions(quantities)}
                  value={quantity}
                  clearable={false}
                  onChange={e =>
                    this.setState({
                      quantity: e.value,
                    })
                  }
                />
              </div>
              <div className="small-12 medium-expand column mb-md">
                <Button
                  className="productProfile__addToBagButton expanded"
                  onClick={() => addToCart(product, quantity)}
                >
                  Add to Cart
                </Button>
              </div>
            </div>
          )}
        </div>
      </div>
    );
  }
}

export default ProductProfile;
