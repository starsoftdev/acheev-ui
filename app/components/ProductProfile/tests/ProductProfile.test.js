import React from 'react';
import { shallow, mount } from 'enzyme';
import { fromJS } from 'immutable';
import { create } from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import ProductProfile from 'components/ProductProfile';
import Button from 'components/Button';

const data = fromJS({
  business: {
    photos: ['https://lift.co/uploads/tLrXUeWNTn1Gq72YtE4d.jpg'],
    hoursOfOperation: [],
    telephone: '1 (855) 558-9333',
    admins: [6266],
    facebook: 'https://www.facebook.com/bedrocancanada',
    createdOn: '2014-12-11T05:06:00.253Z',
    productReviewCount: 1474,
    name: 'Bedrocan',
    slug: 'bedrocan',
    thumbnail: 'https://lift.co/uploads/tLrXUeWNTn1Gq72YtE4d.jpg',
    productRating: 4.5,
    published: true,
    registrationLink: 'https://www.tweedmainstreet.com/account/register',
    twitter: 'https://twitter.com/bedrocancda',
    __t: 'Producer',
    __v: 4,
    reviewCount: 348,
    tags: [],
    _id: '54892638f2c99094235ef27a',
    updatedOn: '2017-08-01T17:19:24.329Z',
    reviews: ['5980b8660952e830dce85c4d', '5980b8660952e830dce85afe'],
    website: 'http://bedrocan.ca',
    id: '54892638f2c99094235ef27a',
    email: 'info@bedrocan.ca',
    description:
      '<p>In 2011, Bedrocan BV, a Dutch company, was invited to Ottawa by Health Canada to comment on Canada’s new regulations for medicinal cannabis, the Marijuana for Medical Purposes Regulations (MMPR).</p>',
    features: {
      promoted: false,
      points: true,
    },
    rating: 4.5,
    linkedin: 'https://www.linkedin.com/company/bedrocan-canada',
    locations: [
      {
        province: 'Ontario',
        city: 'Toronto',
        address: 'PO BOX 89589, 250A Eglinton Ave. East',
        postalCode: 'M4P 1K2',
        _id: '5980b81c0952e830dce61524',
        latLon: [43.7084593, -79.39065719999996],
      },
    ],
  },
  photos: [],
  flavours: [],
  thcLow: 22,
  cbd: null,
  createdOn: '2014-10-01T02:46:14.198Z',
  cbdHigh: null,
  positiveEffects: [],
  symptoms: [],
  name: 'Bedrocan®',
  slug: 'bedrocan-bedrocan',
  negativeEffects: [],
  variants: [
    {
      price: 850,
      doseUnit: 'g',
      doseAmount: 1,
      _id: '5980b81e0952e830dce61863',
      weightUnit: 'g',
      availabilityStatus: 'available',
      available: true,
    },
  ],
  published: true,
  thc: '22%',
  cbdLow: null,
  thcHigh: 22,
  searchField: 'Bedrocan® Bedrocan hybrid',
  prescribedFor: [],
  __t: 'Strain',
  __v: 1,
  awards: [],
  reviewCount: 611,
  title: 'Bedrocan®',
  tags: [],
  _id: '5980b81e0952e830dce61862',
  updatedOn: '2017-05-25T23:25:32.987Z',
  reviews: ['5980b8200952e830dce621be', '5980b8200952e830dce622c4'],
  id: '5980b81e0952e830dce61862',
  milled: false,
  description:
    '<p>Bedrocan® variety is a ‘Sativa’ dominant strain of cannabis characterized by 22% THC and less than 1% CBD (content by weight).</p><p>Bedrocan® was developed in the Netherlands out of a requirement by the Dutch Health Ministry to have a “high THC” variety available to patients. It’s a Sativa plant type, bred because of its high yield, optimal growth characteristics, and derived from the popularly known “Jack Herer” cannabis variety.</p>',
  rating: 4.5,
  category: 'hybrid',
});

const follows = fromJS({
  count: 47,
  isLoading: false,
  error: '',
});
const likes = fromJS({
  count: 52,
  isLoading: false,
  error: '',
});
const product = {
  options: [{ name: 'hey', values: ['hey', 'ho'] }],
};

jest.mock('react-rating', () => 'div');

describe('<ProductProfile />', () => {
  it("shouldn't throw error if data is not specified", () => {
    try {
      shallow(
        <MemoryRouter>
          <ProductProfile />
        </MemoryRouter>
      );
    } catch (error) {
      expect(error).toBeUndefined();
    }
  });

  it("shouldn't throw error if data is specified but follows or likes are not specified", () => {
    try {
      shallow(
        <MemoryRouter>
          <ProductProfile data={data} />
        </MemoryRouter>
      );
    } catch (error) {
      expect(error).toBeUndefined();
    }
  });

  it("shouldn't render Buy Section if shopifyProduct is not specified", () => {
    const renderedComponent = shallow(
      <MemoryRouter>
        <ProductProfile data={data} follows={follows} likes={likes} />
      </MemoryRouter>
    );
    expect(
      renderedComponent.find('.ProductProfile__buySection').length
    ).toEqual(0);
  });

  it('addToCart should be fired when Button is clicked', () => {
    const spy = jest.fn();
    const renderedComponent = mount(
      <MemoryRouter>
        <ProductProfile
          addToCart={spy}
          data={data}
          follows={follows}
          likes={likes}
          shopifyProduct={product}
        />
      </MemoryRouter>
    );
    renderedComponent.find(Button).simulate('click');
    expect(spy).toHaveBeenCalled();
  });

  it('followLikeProduct should be fired when ProductAction is clicked', () => {
    const spy = jest.fn();
    const renderedComponent = mount(
      <MemoryRouter>
        <ProductProfile data={data} currentUser={{}} followLikeProduct={spy} />
      </MemoryRouter>
    );
    renderedComponent
      .find('ProductAction')
      .first()
      .simulate('click');
    expect(spy).toHaveBeenCalled();
  });

  it("shouldn't require auth to follow or like the product when currentUser is specified", () => {
    const renderedComponent = mount(
      <MemoryRouter>
        <ProductProfile data={data} currentUser={{}} />
      </MemoryRouter>
    );
    expect(renderedComponent.find('RequireAuth').length).toEqual(0);
  });

  it('should require auth to follow or like the product when currentUser is not specified', () => {
    const renderedComponent = mount(
      <MemoryRouter>
        <ProductProfile data={data} />
      </MemoryRouter>
    );
    expect(renderedComponent.find('RequireAuth').length).toEqual(2);
  });

  it('renders data if provided', () => {
    const component = create(
      <MemoryRouter>
        <ProductProfile
          data={data}
          follows={follows}
          likes={likes}
          shopifyProduct={product}
        />
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
