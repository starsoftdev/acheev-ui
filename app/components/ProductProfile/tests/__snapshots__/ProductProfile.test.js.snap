// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductProfile /> renders data if provided 1`] = `
<div
  className="productProfile row"
>
  <div
    className="small-12 columns medium-6"
  >
    <div
      className="productProfile__thumbnail mb-md"
      style={
        Object {
          "backgroundImage": "url(https://lift.co/uploads/tLrXUeWNTn1Gq72YtE4d.jpg)",
        }
      }
    />
  </div>
  <div
    className="small-12 columns medium-6"
  >
    <h2
      className="c-darkest-gray"
    >
      Bedrocan®
    </h2>
    <div
      className="productProfile__producerName"
    >
      <a
        className="productProfile__producerNameLink"
        href="/producers/bedrocan"
        onClick={[Function]}
      >
        by 
        Bedrocan
      </a>
    </div>
    <div
      className="row"
    >
      <div
        className="shrink column"
      >
        <div
          className="productCategory"
        >
          <svg
            className="productCategory__icon productCategory__icon--hybrid"
            height={23}
            width={66}
          >
            <use
              xlinkHref="#undefined"
            />
          </svg>
          <div
            className="productCategory__text"
          >
            hybrid
          </div>
        </div>
      </div>
      <div
        className="shrink column"
      >
        <div
          className="cannabinoids row align-center"
        >
          <div
            className="shrink column t-nowrap"
          >
            <strong>
              THC
            </strong>
             
            22%
          </div>
        </div>
      </div>
    </div>
    <div
      className="row align-middle mb-md"
    >
      <div
        className="column small-12 large-expand mb-md"
      >
        <div
          className="reviewCount row align-middle"
        >
          <div
            className="reviewCount__starSection column small-12 medium-shrink"
          >
            <div
              className="reviewCount__starsContainer row align-middle"
            >
              <div
                className="shrink column"
              >
                <div
                  className="starRating reviewCount__star"
                >
                  <div
                    emptySymbol={
                      <Icon
                        glyph="IMAGE_MOCK"
                        size={16}
                      />
                    }
                    fractions={2}
                    fullSymbol={
                      <Icon
                        glyph="IMAGE_MOCK"
                        size={16}
                      />
                    }
                    initialRating={4.5}
                    readonly={true}
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="reviewCount__countSection column small-12 medium-shrink"
          >
            <strong>
              (
              4.5
              )   
            </strong>
            <a
              className=""
              href="/strains/bedrocan-bedrocan/reviews"
              onClick={[Function]}
            >
              2
               
              
              reviews
            </a>
          </div>
        </div>
      </div>
      <div
        className="shrink column mb-md"
      >
        <a
          className="button secondary"
          href="/strains/bedrocan-bedrocan/create-review"
          onClick={[Function]}
        >
          Review
        </a>
      </div>
    </div>
    <div
      className="row align-middle mb-md"
    >
      <div
        className="column"
      >
        <div>
          <span
            className="productProfile__price"
          >
            <span>
              <span>
                $
                8.50
                 
                <span
                  className="t-lowercase"
                >
                  (g)
                </span>
              </span>
            </span>
          </span>
        </div>
      </div>
      <div
        className="shrink column right"
      >
        <div
          onClick={[Function]}
          role="button"
        >
          <div
            className="productAction"
            role="button"
          >
            <svg
              className="productAction__icon"
              height={11}
              width={11}
            >
              <use
                xlinkHref="#undefined"
              />
            </svg>
            Follow
             (
            47
            )
          </div>
        </div>
      </div>
      <div
        className="shrink column"
      >
        <div
          onClick={[Function]}
          role="button"
        >
          <div
            className="productAction"
            role="button"
          >
            <svg
              className="productAction__icon"
              height={11}
              width={11}
            >
              <use
                xlinkHref="#undefined"
              />
            </svg>
            Like
             (
            52
            )
          </div>
        </div>
      </div>
    </div>
    <div
      className="row align-middle mb-sm"
    >
      <div
        className="shrink column npr mb-sm"
      >
        <label
          htmlFor="hey"
        >
          hey
        </label>
      </div>
      <div
        className="column mb-sm"
      >
        <div
          className="reactSelect large"
        >
          <div
            className="Select Select--single"
          >
            <div
              className="Select-control"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
            >
              <div
                className="Select-multi-value-wrapper"
                id="react-select-4--value"
              >
                <div
                  className="Select-placeholder"
                >
                  Select...
                </div>
                <div
                  aria-activedescendant="react-select-4--value"
                  aria-disabled="false"
                  aria-expanded={false}
                  aria-owns=""
                  className="Select-input"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  style={
                    Object {
                      "border": 0,
                      "display": "inline-block",
                      "width": 1,
                    }
                  }
                  tabIndex={0}
                />
              </div>
              <span
                className="Select-arrow-zone"
                onMouseDown={[Function]}
              >
                <svg
                  height={10}
                  width={10}
                >
                  <use
                    xlinkHref="#undefined"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row align-middle"
    >
      <div
        className="shrink column mb-md"
      >
        <label
          htmlFor="qty"
        >
          Quantity
        </label>
      </div>
      <div
        className="small-expand medium-shrink column mb-md"
      >
        <div
          className="reactSelect large"
        >
          <div
            className="Select has-value Select--single"
          >
            <input
              disabled={false}
              name="qty"
              type="hidden"
              value="1"
            />
            <div
              className="Select-control"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
            >
              <div
                className="Select-multi-value-wrapper"
                id="react-select-5--value"
              >
                <div
                  className="Select-value"
                >
                  <span
                    aria-selected="true"
                    className="Select-value-label"
                    id="react-select-5--value-item"
                    role="option"
                  >
                    1
                  </span>
                </div>
                <div
                  aria-activedescendant="react-select-5--value"
                  aria-disabled="false"
                  aria-expanded={false}
                  aria-owns=""
                  className="Select-input"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  style={
                    Object {
                      "border": 0,
                      "display": "inline-block",
                      "width": 1,
                    }
                  }
                  tabIndex={0}
                />
              </div>
              <span
                className="Select-arrow-zone"
                onMouseDown={[Function]}
              >
                <svg
                  height={10}
                  width={10}
                >
                  <use
                    xlinkHref="#undefined"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="small-12 medium-expand column mb-md"
      >
        <a
          className="button productProfile__addToBagButton expanded"
          onClick={[Function]}
        >
          <div
            className="row align-middle align-center"
          >
            <div
              className="shrink column"
            >
              Add to Cart
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
